src/string.cr

```diff
@@ -1173,9 +1173,9 @@ class String
   # "hello".byte_slice(-2, 5)  # => "he"
```

Comment:
This and the two calls above actually return `"lo"`. I don't know why these aren't caught by the routine "fix examples in the docs" PRs

Reply:
@maiha Any idea?

Reply:
The basic premise is that my checkers cannot handle **all** EXAMPLES.
* why: https://github.com/maiha/crystal-examples#difficulty

EXAMPLES that cannot be handled are managed as an "exclusion list" and excluded from testing.
* how: https://github.com/maiha/crystal-examples/blob/master/bundled/heuristic.jnl#L295

In this case, the entire code block is excluded because the incorrect comment syntax would interfere with the automatic generation of tests.
```crystal
"¥hello".byte_slice(0, 1)  # => "\C2" (invalid UTF-8 character)
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^
```

The following code is generated and fails.

```console
$ cd tmp/string/working && LC_ALL=C /v/crystal/bin/crystal run -D preview_overflow ../all_spec.cr 2>&1
Using compiled compiler at /v/crystal/.build/crystal
In /v/tmp/string/all_spec.cr:509:46

 509 | [0m[1m( "¥hello".byte_slice(0, 1) ).should eq( "�" (invalid UTF-8 character) )
                                                    ^
Error: expecting token ')', not '('
```

I understand. So how many examples of such exclusions do you have?
```
1979 test
1341 success 43 pending 3 general 147 pseudo 71 macro 9 wrong 44 random 321 failure
```

Reply:
Perhaps we could fix a format for such comments on the output value.

My initial idea for this would be to just append another comment:
```cr
"¥hello".byte_slice(0, 1)  # => "\C2" # invalid UTF-8 character
```
This should be fairly readable, unambiguous and easy to parse.

Reply:
It's good. I think it is the best choice so far. 💯

Reply:
There should be time. [1.7.0](https://github.com/crystal-lang/crystal/milestone/61) is scheduled for January 6.